### PR TITLE
CDN article

### LINKED ISSUE #2416: content: content delivery networks [2020 May 29]
* Author: @khempenius 
* [Draft](https://docs.google.com/document/d/1JNSpYoIBEBHAUUPdT5Tvxok-sjM3GUjyJl1skfpwfOY/edit?ts=5eaafe81)
* Deadline: Apr 30 (Soft)
* next step: @khempenius to address feedback from 7/29

### PR DESCRIPTION
**Pull request by khempenius**
Fixes #2416

### CODE REVIEW

**Comment by kaycebasques** [comment]
Please re-assign to me when it's ready for review.

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+*   **Freshness**: `Cache-Control: max-age=&lt;seconds>`, `Cache-Control: s-max-age=&lt;seconds>`, or `Expires`.
+
+For a more detailed explanation of HTTP caching, refer to [Prevent unnecessary network requests with the HTTP Cache](https://web.dev/http-cache/).
+
+
+### Fine tuning
+
+A slightly simplified explanation of how CDN caches work is that the URL of a resource is used as the key for caching and retrieving the resource from the cache. In practice, this is still overwhelmingly true, but is complicated slightly by the impact of things like request headers and query params. For example, only using the resource URL as the cache key won't work if a site uses the Accept-Language header for localization. 
+
+Addressing these considerations is not only important for maximizing CHR, but also for ensuring that the correct content is served to users. A properly configured CDN instance strikes the correct balance between overly granular caching (which hurts CHR) and insufficiently granular caching (which results in incorrect responses being served to users).
```
Comment (mnot-fastly) [inline src/site/content/en/fast/content-delivery-networks/index.md:158]: I would state this in terms of rewriting request URLs, rather than rewriting the cache key -- see https://i.blackhat.com/USA-20/Wednesday/us-20-Kettle-Web-Cache-Entanglement-Novel-Pathways-To-Poisoning-wp.pdf

**Review by mnot-fastly — COMMENTED** [review]
Overall this looks really good, left some notes / suggestions for consideration.

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+</figure>
+
+### HTTP/2 & HTTP/3
+
+HTTP/2 and HTTP/3 both provide performance benefits over HTTP/1. Of the two, HTTP/3 offers greater _potential_ performance benefits. HTTP/3 isn't fully standardized yet, but it will be widely [supported](https://caniuse.com/#feat=http3) once this occurs.
+
+
+#### HTTP/2
+
+If your CDN hasn't already enabled [HTTP/2](https://en.wikipedia.org/wiki/HTTP/2) by default, you should consider turning it on. HTTP/2 provides multiple [performance benefits](https://hpbn.co/http2) over HTTP/1 and is [supported](https://caniuse.com/#feat=http2) by all major browsers. Performance features of HTTP/2 include: [multiplexing](https://hpbn.co/http2/#request-and-response-multiplexing), [stream prioritization](https://hpbn.co/http2/#stream-prioritization), [server push](https://developers.google.com/web/fundamentals/performance/http2), and [header compression](https://blog.cloudflare.com/hpack-the-silent-killer-feature-of-http-2/).
```
Comment (dotjs) [inline src/site/content/en/fast/content-delivery-networks/index.md:230]: Maybe use https://almanac.httparchive.org/en/2019/http2 as more up to date review of http/2 and its usage.

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+
+    Multiplexing is arguably the most important feature of HTTP/2. Multiplexing enables a single TCP connection to serve multiple request-response pairs at the same time. This eliminates the overhead of unnecessary connection setups; given that the number of connections that a browser can have open at a given time is limited, this also has the implication that the browser is now able to request more of a page's resources in parallel. Multiplexing theoretically removes the need for HTTP/1 optimizations like concatenation and sprite sheets - however, in practice, these techniques will remain relevant given that larger files compress better.
+
+
+*  **Stream Prioritization**
+
+    Multiplexing enables multiple concurrent streams; [stream prioritization](https://http2.github.io/http2-spec/#StreamPriority) provides an interface for communicating relative priority of each of these streams. This helps a server decide how to allocate its limited resources. Stream prioritization is expressed by the browser via a dependency tree and is merely a statement of _preference_: in other words, the server is not obligated to meet (or even consider) the priorities supplied by the browser.
```
Comment (dotjs) [inline src/site/content/en/fast/content-delivery-networks/index.md:240]: The sequencing of how resources are loaded has a significant impact on how long it takes for a user to see and interact with a page. The ability to use prioritization allows the sever to send the most important resource before less important resources even if the important resource was requested later such as a font. The more of the site that is served through a CDN the more effective the prioritization becomes.

**Comment by kaycebasques** [comment]
@khempenius looks like this is waiting on you to review last round of feedback

**Comment by khempenius** [comment]
@dotjs @mnot-fastly thank you for the feedback!

**Comment by khempenius** [comment]
@kaycebasques ready for your review

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+
+
+<figure class="w-figure">
+  <img src="./cdn1.png" alt="Comparison of connection setup with and without a CDN" class="w-screenshot">
+</figure>
+
+Some CDNs improve upon this even further by routing traffic to the origin through multiple CDN servers spread across the Internet. Connections between CDN servers occur over reliable and highly optimized routes, rather than routes determined by the [Border Gateway Protocol (BGP)](https://en.wikipedia.org/wiki/Border_Gateway_Protocol). Although BGP is the internet's de facto routing protocol, its routing decisions are not always performance oriented. Therefore, BGP-determined routes are likely to be less performant than the finely-tuned routes between CDN servers.
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:36]:
```suggestion
Some CDNs improve upon this even further by routing traffic to the origin through multiple CDN servers spread across the Internet. Connections between CDN servers occur over reliable and highly optimized routes, rather than routes determined by the [Border Gateway Protocol (BGP)](https://en.wikipedia.org/wiki/Border_Gateway_Protocol). Although BGP is the internet's de facto routing protocol, its routing decisions are not always performance-oriented. Therefore, BGP-determined routes are likely to be less performant than the finely-tuned routes between CDN servers.
```

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+CDNs use cache eviction to periodically remove not-so-useful resources from the cache. In addition, site owners can use purging to explicitly remove resources.
+
+
+*  **Cache eviction**
+
+    Caches have a limited storage capacity. When a cache nears its capacity, it makes room for new resources by removing resources that haven't been accessed recently, or which take up a lot of space. This process is known as cache eviction. A resource being evicted from one cache does not necessarily mean that it has been evicted from all caches in a CDN network.
+
+*  **Purging**
+
+    Purging (also known as "cache invalidation") is a mechanism for removing a resource from a CDN's caches without having to wait for it to expire or be evicted. It is typically executed via API. Purging is critical in situations where content needs to be retracted (for example, correcting typos, pricing errors, or incorrect news articles). On top of that, it can also play a crucial role in a site's caching strategy.
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:62]:
```suggestion
    Purging (also known as "cache invalidation") is a mechanism for removing a resource from a CDN's caches without having to wait for it to expire or be evicted. It is typically executed via an API. Purging is critical in situations where content needs to be retracted (for example, correcting typos, pricing errors, or incorrect news articles). On top of that, it can also play a crucial role in a site's caching strategy.
```

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+
+    If a CDN supports near instant purging, purging can be used as a mechanism for managing the caching of dynamic content: cache dynamic content using a long TTL, then purge the resource whenever it is updated. In this way, it is possible to maximize the caching duration of a dynamic resource, despite not knowing in advance when the resource will change. This technique is sometimes referred to as "hold-till-told caching".
+
+    When purging is used at scale it is typically used in conjunction with a concept known as "cache tags" or "surrogate cache keys". This mechanism allows site owners to associate one or more additional identifiers (sometimes referred to as "tags") with a cached resource. These tags can then be used to carry out highly granular purging. For example, you might add a "footer" tag to all resources (for example, "/about", "/blog") that contain your site footer . When the footer is updated, instruct your CDN to purge all resources associated with the "footer" tag.
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:66]:
```suggestion
    When purging is used at scale it is typically used in conjunction with a concept known as "cache tags" or "surrogate cache keys". This mechanism allows site owners to associate one or more additional identifiers (sometimes referred to as "tags") with a cached resource. These tags can then be used to carry out highly granular purging. For example, you might add a "footer" tag to all resources (for example, `/about`, `/blog`) that contain your site footer. When the footer is updated, instruct your CDN to purge all resources associated with the "footer" tag.
```

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+
+#### Cacheable Resources
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:68]:
```suggestion
#### Cacheable resources
```

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+
+*  **Private Resources**
+
+    Private resources contain data intended for a single user and therefore should not be cached by a CDN. Private resources are indicated by the `Cache-Control: private` header.
+
+*  **Public Resources**
+
+    Public resources do not contain user-specific information and therefore are cacheable by a CDN. A resource is considered cacheable by a CDN if it does not have a `Cache-Control: no-store` or `Cache-Control: private` header. The length of time that a public resource can be cached depends on how frequently the asset changes.
+
+##### Dyanmic and static content
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:82]:
```suggestion
##### Dynamic and static content  
```

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+
+
+### Additional features
+
+CDNs typically offer a wide variety of features in addition to their core CDN offering. Commonly offered features include: load balancing, image optimization, video streaming, edge computing, and security products.
+
+
+## How to setup and configure a CDN
+
+Ideally a CDN should be used to serve your entire site. At a high-level, the setup process for this consists of signing up with a CDN provider, then updating your CNAME DNS record to point at the CDN provider. For example, the CNAME record for `www.example.com` might point to `example.my-cdn.com`. As a result of this DNS change, traffic to your site will be routed through the CDN.
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:115]:
```suggestion
Ideally you should use a CDN to serve your entire site. At a high-level, the setup process for this consists of signing up with a CDN provider, then updating your CNAME DNS record to point at the CDN provider. For example, the CNAME record for `www.example.com` might point to `example.my-cdn.com`. As a result of this DNS change, traffic to your site will be routed through the CDN.
```
Rephrasing the first sentence to avoid passive voice

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+
+If using a CDN to serve all resources is not an option, a CDN can be configured to only serve a subset of resources - for example, only static resources. This is accomplished by creating a separate CNAME record that will only be used for resources that should be served by the CDN. For example, creating a `static.example.com` CNAME record that points to `example.my-cdn.com`. The URLs of resources that should be served by the CDN should be rewritten to point at the `static.example.com` subdomain that you just created.
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:117]:
```suggestion
If using a CDN to serve all resources is not an option, you can configure a CDN to only serve a subset of resources - for example, only static resources. You can do this by creating a separate CNAME record that will only be used for resources that should be served by the CDN. For example, you might create a `static.example.com` CNAME record that points to `example.my-cdn.com`. You would also need to rewrite the URLs of resources being served by the CDN to point to the `static.example.com` subdomain that you created.
```
This is an example of why the passive voice is important to avoid. When we rephrase it to active voice, it makes it clearer that we have to do all of these steps; the CDN isn't going to do them for us

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+Although your CDN will be set up at this point, there will likely be inefficiencies in your configuration. The next two sections of this article will explain how to get the most out of your CDN by increasing cache hit ratio and enabling performance features.
+
+
+## Improving cache hit ratio
+
+An effective CDN setup will serve as many resources as possible from the cache. This is commonly measured by cache hit ratio (CHR). Cache hit ratio is defined as the number of cache hits divided by the number of total requests during a given time interval.
+
+A freshly initialized cache will have a CHR of 0 but this increases as the cache is populated with resources. A CHR of 90% is a good goal for most sites. 
+
+When optimizing CHR, the first thing to verify is that all cacheable resources are being cached and cached for the correct length of time. This is a simple assessment that should be undertaken by all sites.
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:128]: What makes it simple? Is it because all good CDNs provide this feature? If that's the case, I think we should mention something along those lines. In other words we should make it clear that it's a red flag if a supposed CDN doesn't provide CHR data.

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 12]
```diff
@@ -0,0 +1,284 @@
...
+
+The next level of CHR optimization, broadly speaking, is to fine tune your CDN settings to make sure that logically equivalent server responses aren't being cached separately. This is a common inefficiency that occurs due to the impact of factors like query params, cookies, and request headers on caching.
+
+
+### Initial Audit
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:133]:
```suggestion
### Initial audit
```

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 13]
```diff
@@ -0,0 +1,284 @@
...
+
+For a more detailed explanation of HTTP caching, refer to [Prevent unnecessary network requests with the HTTP Cache](https://web.dev/http-cache/).
+
+
+### Fine tuning
+
+A slightly simplified explanation of how CDN caches work is that the URL of a resource is used as the key for caching and retrieving the resource from the cache. In practice, this is still overwhelmingly true, but is complicated slightly by the impact of things like request headers and query params. As a result, rewriting request URLs is an important technique for both maximizing CHR and ensuring that the correct content is served to users. A properly configured CDN instance strikes the correct balance between overly granular caching (which hurts CHR) and insufficiently granular caching (which results in incorrect responses being served to users).
+
+#### Query Params
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:152]:
```suggestion
#### Query params
```

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 14]
```diff
@@ -0,0 +1,284 @@
...
+
+By default, CDNs take query params into consideration when caching a resource. However, small adjustments to query param handling can have a significant impact on CHR. For example:
+
+
+
+*   **Unnecessary query params**
+
+    By default, a CDN would cache `example.com/blog` and `example.com/blog?referral\_id=2zjk` separately even though they are likely the same underlying resource. This is fixed by adjusting a CDN's configuration to ignore the `referral\_id` query param.
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:160]: Is the backslash (`referral\_id`) intentional?

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 15]
```diff
@@ -0,0 +1,284 @@
...
+
+
+## Performance features
+
+This section discusses performance features that are commonly offered by CDNs as part of their core product offering. Many sites forget to enable these features, thereby losing out on easy performance wins.
+
+
+### Compression
+
+All text-based responses should be compressed with either gzip or Brotli. If you have the choice, choose Brotli over gzip. Brotli is a newer compression algorithm, and compared to gzip, it can achieve higher compression ratios.
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:186]:
```suggestion
All text-based responses should be [compressed](/reduce-network-payloads-using-text-compression/#data-compression) with either gzip or Brotli. If you have the choice, choose Brotli over gzip. Brotli is a newer compression algorithm, and compared to gzip, it can achieve higher compression ratios.  
```

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 16]
```diff
@@ -0,0 +1,284 @@
...
+
+
+#### Compression best practices
+
+Sites that want to maximize performance should apply Brotli compression at both their origin server and CDN. Brotli compression at the origin minimizes the transfer size of resources that can't be served from the cache. To prevent delays in serving requests, the origin should compress dynamic resources using a fairly conservative compression level - for example, Brotli-4; static resources can be compressed using Brotli-11.
+
+
+### TLS 1.3
+
+TLS 1.3 is the newest version of [Transport Layer Security (TLS)](https://en.wikipedia.org/wiki/Transport_Layer_Security), the cryptographic protocol used by [HTTPS](https://en.wikipedia.org/wiki/HTTPS). TLS 1.3 provides better privacy and performanced compared to TLS 1.2
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:212]:
```suggestion
TLS 1.3 is the newest version of [Transport Layer Security (TLS)](https://en.wikipedia.org/wiki/Transport_Layer_Security), the cryptographic protocol used by [HTTPS](https://en.wikipedia.org/wiki/HTTPS). TLS 1.3 provides better privacy and performance compared to TLS 1.2.
```

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 17]
```diff
@@ -0,0 +1,284 @@
...
+
+TLS 1.3. shortens the TLS handshake from two roundtrips to one. For connections using HTTP/1 or HTTP/2, shortening the TLS handshake to one roundtrip effectively reduces connection setup time by 33%.
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:214]:
```suggestion
TLS 1.3 shortens the TLS handshake from two roundtrips to one. For connections using HTTP/1 or HTTP/2, shortening the TLS handshake to one roundtrip effectively reduces connection setup time by 33%.
```

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 18]
```diff
@@ -0,0 +1,284 @@
...
+
+<figure class="w-figure">
+  <img src="./cdn3.png" alt="Comparison of the TLS 1.2 and TLS 1.3 handshakes" class="w-screenshot">
+</figure>
+
+### HTTP/2 & HTTP/3
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:220]:
```suggestion
### HTTP/2 and HTTP/3
```

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 19]
```diff
@@ -0,0 +1,284 @@
...
+
+HTTP/2 and HTTP/3 both provide performance benefits over HTTP/1. Of the two, HTTP/3 offers greater _potential_ performance benefits. HTTP/3 isn't fully standardized yet, but it will be widely [supported](https://caniuse.com/#feat=http3) once this occurs.
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:222]: Is there a reason that we are making this strong claim about HTTP/3? It seems risky to say "it **will** be supported once it gets fully standardized" especially since there are factors outside of our control (other browser vendors)

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 20]
```diff
@@ -0,0 +1,284 @@
...
+
+*  **Multiplexing**
+
+    Multiplexing is arguably the most important feature of HTTP/2. Multiplexing enables a single TCP connection to serve multiple request-response pairs at the same time. This eliminates the overhead of unnecessary connection setups; given that the number of connections that a browser can have open at a given time is limited, this also has the implication that the browser is now able to request more of a page's resources in parallel. Multiplexing theoretically removes the need for HTTP/1 optimizations like concatenation and sprite sheets - however, in practice, these techniques will remain relevant given that larger files compress better.
+
+
+*  **Stream Prioritization**
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:234]:
```suggestion
*  **Stream prioritization**
```

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 21]
```diff
@@ -0,0 +1,284 @@
...
+
+Stream prioritization is expressed by the browser via a dependency tree and is merely a statement of _preference_: in other words, the server is not obligated to meet (or even consider) the priorities supplied by the browser. Stream prioritization becomes more effective when more of a site is served through a CDN.
+
+CDN implementations of HTTP/2 resource prioritization vary wildly. To identify whether your CDN fully and properly supports HTTP/2 resource prioritization, check out [Is HTTP/2 Fast Yet?](https://ishttp2fastyet.com/).
+
+
+Although switching your CDN instance to HTTP/2 is largely a matter of flipping a switch, it's important to thoroughly test this change before enabling it in production. HTTP/1 and HTTP/2 use the same conventions for request and response headers - but HTTP/2 is far less forgiving when these conventions aren't adhered to. As a result, non-spec practices like including non-ASCII or uppercase characters in headers may begin causing errors once HTTP/2 is enabled.
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:243]: Any links/resources on how to verify that migration from HTTP/1 to HTTP/2 has been successful?

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 22]
```diff
@@ -0,0 +1,284 @@
...
+
+
+### Image optimization
+
+CDN image optimization services typically focus on image optimizations that can be applied automatically in order to reduce image transfer size. For example: stripping [EXIF](https://en.wikipedia.org/wiki/Exif) data, applying lossless compression, and converting images to newer file formats (for example, WebP). Images make up ~50% of the transfer bytes on the median web page, so optimizing images can significantly reduce page size.
+
+
+### Minification
+
+Minification removes unnecessary characters from JavaScript, CSS, and HTML. It's preferable to do minification at the origin server, rather than the CDN. Site owners have more context about the code to be minified and therefore can often use more aggressive minification techniques than those employed by CDNs. However, if minifying code at the origin is not an option, minification by the CDN is a good alternative.
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:277]:
```suggestion
[Minification](/reduce-network-payloads-using-text-compression/#minification) removes unnecessary characters from JavaScript, CSS, and HTML. It's preferable to do minification at the origin server, rather than the CDN. Site owners have more context about the code to be minified and therefore can often use more aggressive minification techniques than those employed by CDNs. However, if minifying code at the origin is not an option, minification by the CDN is a good alternative.
```

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 23]
```diff
@@ -0,0 +1,284 @@
...
+
+
+## Conclusion
+
+*   **Use a CDN:** CDNs deliver resources quickly, reduce load on the origin server, and are helpful for dealing with traffic spikes.
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:282]: On my first readthrough I didn't see much about the idea that "CDNs reduce load on the origin server" and I think it's a good one (because it might save someone money). You might want to go back and see if there's anywhere you can expand on this idea (even if it's only a sentence, and the sentence is very direct, e.g. "a CDN may save you money"). On the other hand you might already cover it well and my reading comprehension just wasn't that thorough.

**Review by kaycebasques — CHANGES_REQUESTED** [review]
Looking good, lots of in-depth ideas that I've never seen before

Tip: Use the [batch commit suggestions](https://groups.google.com/a/google.com/g/web-devrel-chrome-team/c/g6dEuRQy8Fc/m/99N2tw5IBwAJ) workflow to incorporate my requested changes faster.

Please break all lines at 100 characters. If you're using VS Code, [Rewrap](https://marketplace.visualstudio.com/items?itemName=stkb.rewrap) makes it pretty easy to do. The [Docs To X](https://gsuite.google.com/u/0/marketplace/app/docs_to_x/243920501110?hl=en&pann=docs_addon_widget) Google Docs Add-On also supports this. One reason is because GitHub's UI isn't good at formatting requested changes when lines are long:

image

I noticed a few instances of the passive voice and may want to read again and see if I missed any

We should link to this from web.dev/fast, yeah?

(The requested changes below were auto-generated by the web.dev content review extension)

- [ ] Resize the hero image: https://web.dev/handbook/markup-media/#hero
- [ ] Resize the width of the following images to 1600px or less:
  - [ ] cdn2.png
- [ ] Add a subheading to the page: https://web.dev/handbook/yaml-front-matter/#subhead

File: src/site/content/en/fast/content-delivery-networks/index.md (hunk 19 above)
Comment (khempenius) [reply src/site/content/en/fast/content-delivery-networks/index.md:226]: This should be OK. H3 is in all major browsers at the moment - albeit behind a flag. My original wording was less strong than this but it was suggested that this would be more accurate.

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 24]
```diff
@@ -0,0 +1,290 @@
...
+
+## Overview
+
+A content delivery network consists of a network of servers that are optimized for quickly delivering content to users. Although CDNs are arguably best known for serving cached content, CDNs can also improve the delivery of uncacheable content. Generally speaking, the more of your site delivered by your CDN, the better.
+
+At a high-level, the performance benefits of CDNs stem from a handful of principles: CDN servers are located closer to users than [origin servers](https://en.wikipedia.org/wiki/Upstream_server) and therefore have a shorter [round-trip time (RTT)](https://en.wikipedia.org/wiki/Round-trip_delay) latency; networking optimizations allow CDNs to deliver content more quickly than if the content was loaded "directly" from the origin server; lastly, CDN caches eliminate the need for a request to travel to the origin server.
+
+{% Aside %}
+"Origin server" refers to the server that a CDN retrieves content from.
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:28]:
```suggestion
{% Aside 'key-term' %}
**Origin server** refers to the server that a CDN retrieves content from.
```

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 25]
```diff
@@ -0,0 +1,290 @@
...
+title: Content delivery networks (CDNs)
+authors:
+  - katiehempenius
+description: |
+  This article provides a comprehensive overview of content delivery networks (CDNs). 
+  In addition, it explains how to choose, configure, and optimize a CDN setup.
+subhead: |
+  Improve performance by using a content delivery network. 
+date: 2020-09-14
+updated: 2020-09-14
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:12]:
```suggestion
date: 2020-09-21
#updated: 2020-09-14
```

**Review by kaycebasques — APPROVED** [review]
LGTM. I'll merge this Tuesday 9/21.

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 26]
```diff
@@ -0,0 +1,290 @@
...
+layout: post
+title: Content delivery networks (CDNs)
+authors:
+  - katiehempenius
+description: |
+  This article provides a comprehensive overview of content delivery networks (CDNs). 
+  In addition, it explains how to choose, configure, and optimize a CDN setup.
+subhead: |
+  Improve performance by using a content delivery network. 
+date: 2020-09-21
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:11]:
```suggestion
date: 2020-09-22
```

File: src/site/content/en/fast/content-delivery-networks/index.md [hunk 27]
```diff
@@ -0,0 +1,290 @@
...
+  In addition, it explains how to choose, configure, and optimize a CDN setup.
+subhead: |
+  Improve performance by using a content delivery network. 
+date: 2020-09-22
+#updated: 2020-09-14
+hero: hero.jpg
+tags:
+  - blog
+  - performance
+---
```
Comment (kaycebasques) [inline src/site/content/en/fast/content-delivery-networks/index.md:17]:
```suggestion
tags:
  - blog
  - performance
  - network
---
```

**Review by kaycebasques — APPROVED** [review]
